### DISCUSSION TITLE
No support for IDN domainnames in VIRTUALHOST environment

### DISCUSSION THREAD

**Discussion by osos** [nginx-proxy/nginx-proxy#1864] [Ideas]
Having virtualhost containing special characters (aka æøåöäü) in the name are not translated with punycode, thus resulting in nginx not serving the site.

I suggest that if a domain containing other characters than [a-z0-9] they are converted to punycode domains.

See converter here: https://www.idnconverter.se/

Info: https://en.wikipedia.org/wiki/Internationalized_domain_name

Thanks

**Comment by buchdag** [comment]
Hi. It think this is doable, but the unicode to punycode conversion would need to be implemented in [nginx-proxy/docker-gen](https://github.com/nginx-proxy/docker-gen), which does the actual processing of the templates.

**Reply by osos** [reply]
Would make sense. Also the conversion to punycode would always make sense as the the domains are otherwise void as they cannot be resolved.

Not sure how the SSL scripts handle it, but converting to punycode prior to requesting the SSL / acme-challenge should ensure that the acme-script etc can just handle as usual.

**Reply by buchdag** [reply]
acme-companion also use docker-gen to generate its internal config file, so the feature being added to docker-gen would benefit both with a similar set of modifications.
